### PR TITLE
KN-15608 MemCached -> Redis

### PR DESCRIPTION
**Pull request by AlexPhilippov**

### CODE REVIEW

File: Core/LocalizationCache.cs [hunk 1]
```diff
@@ -7,50 +7,49 @@ public static class LocalizationCache
 	{
 		private static ConcurrentDictionary<string, object> _memoryCache = new ConcurrentDictionary<string, object>();
 		static ILocalizationCache _cache;
+
+		const string LocalizationChannel = "localization";
 		public static void Initialize(ILocalizationCache cache)
 		{
 			_cache = cache;
+			_cache.Subscribe(LocalizationChannel, OnClearKey);
```
Comment (polyakova) [inline Core/LocalizationCache.cs]: how this method should be implemented if i want use HttpCache?

File: Core/LocalizationCache.cs [hunk 2]
```diff
@@ -7,50 +7,49 @@ public static class LocalizationCache
...
 
 		public static void Insert<T>(string key, T value)
 		{
-			var timeStamp = DateTime.UtcNow;
-			_cache.Set("stamp_" + key, timeStamp, timeStamp.AddDays(1));
-			_cache.Set(key, value, timeStamp.AddDays(1));
-			_memoryCache["stamp_" + key] = timeStamp;
+			_cache.Set(key, value, DateTime.UtcNow.AddDays(1));
 			_memoryCache[key] = value;
+			_cache.Publish(LocalizationChannel, key);
```
Comment (polyakova) [inline Core/LocalizationCache.cs]: how this method should be implemented if i want use HttpCache?

File: Core/LocalizationCache.cs (hunk 1 above)
Comment (AlexPhilippov) [inline Core/LocalizationCache.cs]: Since it in-memory cache, we just add list of callbacks and call them

File: Sample/HttpCache.cs [hunk 3]
```diff
@@ -17,32 +17,53 @@ public HttpCache()
...
+		public void Clear(string region = null)
 		{
 			var items = _cache.GetEnumerator();
 
 			while (items.MoveNext())
 				_cache.Remove(items.Key.ToString());
 		}
+
+		const string ChannelPrefix = "channel_";
+		public void Subscribe(string channel, Action<string, string> callback)
```
Comment (polyakova) [inline Sample/HttpCache.cs]: why is this method needed in HttpCache implementation? If i download library from nuget should i do all these things in trivial implementation? I think it become a bit complicated

Comment (AlexPhilippov) [inline Sample/HttpCache.cs]: This allows us to remove all timestamps-related staff. So we need it in trivial implementation.

File: Core/LocalizationCache.cs (hunk 1 above)
Comment (polyakova) [inline Core/LocalizationCache.cs]: This is an open source lib and i don't think that most of people use redis or something like that. The interface should be as much simple as possible i think.

File: Sample/HttpCache.cs [hunk 4]
```diff
@@ -11,38 +11,68 @@ namespace Sample
...
+		{
+			var channelKey = ChannelPrefix + channel;
+			List<Action<string, string>> callbacks;
+			_callbacks.TryGetValue(channelKey, out callbacks);
+			if (callbacks == null)
+				callbacks = new List<Action<string, string>>();
+			callbacks.Add(callback);
+			_callbacks[channelKey] = callbacks;
```
Comment (polyakova) [inline Sample/HttpCache.cs]: i think better to check for Key before:
if (_callbacks.Keys.Contains(key)) ...

Comment (AlexPhilippov) [inline Sample/HttpCache.cs]: Operator[] replaces or adds new element by key. Method Add would generate exception and will require such check, that's why I'm using operator[].

**Comment by AlexPhilippov** [comment]
Close to add the same with latest master changes

**Pull request closed without merging**